packages/contracts/contracts/split/split-provider/SplitProviderV0.sol

```diff
@@ -0,0 +1,190 @@
+        bytes4 interfaceId
+    ) public view override returns (bool) {
+        // broadcast support of the getOrCreateSplitter function
+        // @dev intentionally only including getOrCreateSplitter, as that is the only function
+        // that is intended to be used for on-chain integration (_type is for indexing and introspection only)
+        return
+            interfaceId == this.getOrCreateSplitter.selector ||
+            super.supportsInterface(interfaceId);
+    }
+
+    /**
+     * @notice Indicates the type of the contract, e.g. `SplitProviderV0`.
+     * @return type_ The type of the contract.
+     */
+    function type_() external pure returns (bytes32) {
+        return TYPE;
+    }
+
+    function _getSplitParams(
+        SplitInputs calldata splitInputs
+    ) private pure returns (ISplitFactoryV2.Split memory splitParams) {
+        // define allocations in BPS to avoid any rounding errors
+        uint256 ArtistAndAdditionalRoyaltyPercentage = splitInputs
+            .artistTotalRoyaltyPercentage;
+        uint256 ArtistAndAdditionalRoyaltyBPS = 100 *
+            ArtistAndAdditionalRoyaltyPercentage;
+        // load render provider royalty BPS
+        uint256 renderProviderRoyaltyBPS = splitInputs
+            .renderProviderSecondarySalesBPS;
+        // load platform provider royalty BPS
+        uint256 platformProviderRoyaltyBPS = splitInputs
+            .platformProviderSecondarySalesBPS;
+
+        // edge case: no royalties due; revert
+        if (
+            ArtistAndAdditionalRoyaltyPercentage +
+                renderProviderRoyaltyBPS +
+                platformProviderRoyaltyBPS ==
+            0
+        ) {
+            revert("SplitProviderV0: No royalties due");
+        }
+
+        bool artistNonZero = ArtistAndAdditionalRoyaltyPercentage > 0 &&
+            splitInputs.additionalPayeePercentage < 100;
+        bool additionalPayeeNonZero = ArtistAndAdditionalRoyaltyPercentage >
+            0 &&
+            splitInputs.additionalPayeePercentage > 0;
+        bool renderNonZero = renderProviderRoyaltyBPS > 0;
+        bool platformNonZero = platformProviderRoyaltyBPS > 0;
+
+        // allocate memory for splitParams arrays
+        uint256 partyLength = (artistNonZero ? 1 : 0) +
+            (additionalPayeeNonZero ? 1 : 0) +
+            (renderNonZero ? 1 : 0) +
+            (platformNonZero ? 1 : 0);
+        splitParams.recipients = new address[](partyLength);
+        splitParams.allocations = new uint256[](partyLength);
+        splitParams.totalAllocation =
+            ArtistAndAdditionalRoyaltyBPS +
+            renderProviderRoyaltyBPS +
+            platformProviderRoyaltyBPS;
+
+        // populate each party's split
+        uint256 currentIndex = 0;
+        if (artistNonZero) {
+            splitParams.recipients[currentIndex] = splitInputs.artist;
+            // @dev percent * 100 converts to BPS
+            splitParams.allocations[currentIndex++] =
```

Comment:
would the allocation be:

= (ArtistAndAdditionalRoyaltyPercentage - splitInputs.additionalPayeePercentage) * 100?

Reply:
in this case, since "allocation" is an arbitrary value that only defines how royalties are split, we are setting allocations equal to our finest-grained measure of royalty allocation, Basis Points, to prevent any rounding errors.


Basis points are equal to `percent * 100`, so in this case, we can say:
```
totalArtistAndAdditionalAllocation = artistAndAdditionalRoyaltyPercentage * 100;
```
However, we need to further break down artist and additional's split between that total:
```
artistAllocation = totalArtistAndAdditionalAllocation * ((100-additionalPayeePercentage) / 100);
additionalPayeeAllocation = totalArtistAndAdditionalAllocation * (additionalPayeePercentage / 100);
```
>note: the above have major int division rounding errors, do not use directly!

combining those two equations and cancelling out (100/100) gives:
```
artistAllocation = artistAndAdditionalRoyaltyPercentage * (100 - additionalPayeePercentage);
additionalPayeeAllocation = artistAndAdditionalRoyaltyPercentage * additionalPayeePercentage;
```

...which also happen to be free from any rounding errors, and very gas efficient 😄 

I think this is correct, but definitely let me know if it still looks wrong. I'm bad at percentages 💀

Reply:
Nice, that makes sense to me! Thanks for the explanation 🙏